x-pack/test/security_solution_api_integration/test_suites/detections_response/utils/retry_delete_by_query_conflicts.ts

```diff
@@ -18,34 +18,29 @@ const RETRY_DELAY = 200;
  * Retry an Elasticsearch deleteByQuery operation if it runs into 409 Conflicts,
  * up to a maximum number of attempts.
  */
-export async function retryIfDeleteByQueryConflicts<T>(
+export async function retryIfDeleteByQueryConflicts(
   logger: ToolingLog,
   name: string,
   operation: () => Promise<DeleteByQueryResponse>,
   retries: number = RETRY_ATTEMPTS,
   retryDelay: number = RETRY_DELAY
 ): Promise<DeleteByQueryResponse> {
-  const operationResult = await operation();
-  if (!operationResult.failures || operationResult.failures?.length === 0) {
-    return operationResult;
-  }
+  for (let retriesLeft = retries; retriesLeft > 0; retriesLeft--) {
+    const operationResult = await operation();
```

Comment:
Can `operation` throw an exception? Should we add a try-catch block and retry when it throws?

Reply:
Yes, in a general case it could throw an exception. But I don't think we need to overcomplicate it and add try/catch with logging since unhandled exceptions will be handled globally and shown in the console.

As an example I added exception throwing with 50% chance and ran flaky test runner. You can check how [it looks like](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/6915#0191dfe2-0a6b-4ad6-9dc2-45d41f937c98).